### ISSUE TITLE
Unable to debug azure functions on Mac OS 

### ISSUE THREAD

**Issue by HOllarves** [microsoft/vscode-azurefunctions#2074]
OS: OSX Catalina 10.15.4

Vscode info:
Version: 1.44.2
Commit: ff915844119ce9485abfe8aa9076ec76b5300ddd
Date: 2020-04-16T17:07:18.473Z (2 wks ago)
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 19.4.0

Hey guys! Since today I'm unable to debug azure functions. What I mean is that breakpoints are not hitting, so I'm unable to inspect my code when running it. I haven't changed or installed any new extension. 

launch.json:

```
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Attach to Node Functions",
      "type": "node",
      "request": "attach",
      "port": 9229,
      "preLaunchTask": "func: host start"
    }
  ]
}
```

tasks.json

```
{
    "version": "2.0.0",
    "tasks": [
        {
            "type": "func",
            "command": "host start",
            "problemMatcher": "$func-watch",
            "isBackground": true,
            "dependsOn": "npm install"
        },
        {
            "type": "shell",
            "label": "npm install",
            "command": "npm install"
        },
        {
            "type": "shell",
            "label": "npm prune",
            "command": "npm prune --production",
            "problemMatcher": []
        }
    ]
}
```

settings.json

```
{
  "azureFunctions.deploySubpath": ".",
  "azureFunctions.projectLanguage": "JavaScript",
  "azureFunctions.projectRuntime": "~2",
  "azureFunctions.preDeployTask": "npm prune",
  "editor.formatOnSave": true
}
```

I'm getting no breakpoint hits and callstack, watch, and variables sections remain empty. Breakpoint dots are solid red with no warnings being shown in the terminal, output or debug console tabs.

If I enable the "All exceptions" option I get an infinite loop of exceptions in the file `worker-bundle.js` on line 7130. Don't know if that is expected:

<img width="1133" alt="Screenshot 2020-05-05 at 14 58 07" src="https://user-images.githubusercontent.com/10587409/81068570-dffdeb80-8ee0-11ea-84f8-16f6034b2426.png">

Any ideas? Could it be a recent update?

**Comment by EricJizbaMSFT** [comment]
Hmm well we haven't updated the Functions extension in about a month and the func cli hasn't been updated in even longer, so probably not that. Your settings all look fine - can you copy the logs from the "func host start" task? Also is this project on GitHub and could we take a look?

**Comment by HOllarves** [comment]
Hey @EricJizbaMSFT thanks for the reply

I believe I know what's wrong. Let me provide a bit more context:

I'm building an Apollo federated graph (which is a fancy way to refer to a bunch of GraphQL APIs working together), the problem started in the main endpoint of the graph, which works as a gateway for clients to communicate with all the APIs. 

So the thing is that I realized that if I went to a specific GraphQL service url, and tried to perform a query, breakpoints did work! They just won't work when making queries from the gateway.

So I started to dig a little bit, and I realized that in the "Loaded scripts" section of the debugger, all the GraphQL APIs are listed, EXCEPT for the gateway. 

Here we can see the loaded scripts and the folders they're coming from, all the services listed in the gateway are listed except the gateway itself.
<img width="1741" alt="Screenshot 2020-05-06 at 13 06 11" src="https://user-images.githubusercontent.com/10587409/81170152-77734500-8f9a-11ea-9551-393ccd1e329e.png">

Here are the files and folders present in the project. The gateway is in the `GraphQL` folder.
<img width="475" alt="Screenshot 2020-05-06 at 13 08 57" src="https://user-images.githubusercontent.com/10587409/81170331-c5884880-8f9a-11ea-839a-c43f4ac8cd1e.png">

So, how do I make sure the GraphQL folder gets loaded by the debugger?

Thanks!

**Comment by joeyslack** [comment]
Having the exact same issue - however not related to graph.

Azure Blob Trigger (in) with pattern matching, and javascript (node) runtime, with no functionality, just empty promise returned.

I get the exact same JSON tokenization error.

**Comment by EricJizbaMSFT** [comment]
@HOllarves is the gateway thing running as a separate node process? If so - that would explain why it's not loaded. Is the project on GitHub by chance? Would be easier to understand what's going on if I could see the code. If you don't want the whole project on GitHub, you could also try to make a small sample project that reproduces the same problem

@joeyslack if you take a closer look at your screenshot, the error is caught and ignored a few lines later. I can reproduce if I turn on "All exceptions", but the error doesn't bubble up or prevent me from hitting my own breakpoint for a simple blob trigger. Generally I keep "All exceptions" turned off because it can get pretty noisy

![Screen Shot 2020-05-07 at 12 01 35 PM](https://user-images.githubusercontent.com/11282622/81334185-85889a80-905a-11ea-86d9-097382d78ac2.png)

> latest brew release is 3.0.2245, but github says your newest release was in March, any plans to update homebrew at any time?

I have no idea why there's a discrepancy. The core-tools repo you mentioned is the right place to follow up. Looks like there's at least one issue on this already: https://github.com/Azure/azure-functions-core-tools/issues/1901

**Comment by HOllarves** [comment]
Hey guys! I'm closing this issue because I realized it was an error coming from my side of things. It was one of those stupid simple errors that somehow remain hidden at plain sight. 

My problem was that the gateway was listening to EXTERNAL services. I simply forgot to change switch to my local .env file. No breakpoints were ever hit because the execution was taking place in Azure Cloud and not in my local computer. 

It was working with the other services because when I interacted with them I was doing it directly in my local machine.

Thanks for your help @EricJizbaMSFT !

@joeyslack you might double check the execution of the code is taking place in your local machine :)
